packages/app/src/explorer/utils.ts

```diff
@@ -1,21 +1,23 @@
-import type { Entity } from '@h5web/shared';
-import { assertStr, isGroup } from '@h5web/shared';
+import type { AttributeValues, Entity, Group } from '@h5web/shared';
+import { assertStr } from '@h5web/shared';
+import type { FetchStore } from 'react-suspense-fetch';
 
-import { getAttributeValue, hasAttribute } from '../utils';
+import { hasAttribute } from '../utils';
 
 const SUPPORTED_NX_CLASSES = new Set(['NXdata', 'NXentry', 'NXprocess']);
 
-export function isNxGroup(entity: Entity): boolean {
-  if (!isGroup(entity)) {
-    return false;
-  }
-
-  if (hasAttribute(entity, 'default')) {
+export function needsNxBadge(
+  group: Group,
+  attrValuesStore: FetchStore<AttributeValues, Entity>
+): boolean {
+  if (hasAttribute(group, 'default')) {
     return true;
   }
 
-  if (hasAttribute(entity, 'NX_class')) {
-    const nxClass = getAttributeValue(entity, 'NX_class');
+  if (hasAttribute(group, 'NX_class')) {
+    const attrValues = attrValuesStore.get(group);
+    const nxClass = attrValues?.NX_class;
```

Comment:
At some point, we may put this inside a `getAttributeValue` function.

Reply:
Yeah, I'd like to improve the stores' APIs.